### PR TITLE
Add Route struct getters

### LINKED ISSUE #194: Expose Route internals
## Is your feature request related to a problem? Please describe

As from 0.32 version [`Route`](https://github.com/beatlabs/patron/blob/aa707fa3c1d1a6f2c4ab6fb6e518a7e89c033d4a/component/http/route.go#L14) struct internals are completely hidden from other packages, which leads to overcomplicated or even impossible upgrade from older versions. Existing route tests are mostly affected.

## Describe the solution

- Add getters for `Route` structure internals
- Add method to retrieve `http.HandlerFunc` compliant handler

### PR DESCRIPTION
**Pull request by Oberonus**
## Which problem is this PR solving?

Closes https://github.com/beatlabs/patron/issues/194

## Short description of the changes

Adds getters to all internals of Route struct

### CODE REVIEW

File: component/http/route.go [hunk 1]
```diff
@@ -18,6 +18,26 @@ type Route struct {
 	middlewares []MiddlewareFunc
 }
 
+// GetPath returns route path value.
+func (r Route) GetPath() string {
```
Comment (mantzas) [inline component/http/route.go]: The convention in Go for getter is to omit the Get part and fix the comment.
```suggestion
func (r Route) Path() string {
```

File: component/http/route.go [hunk 2]
```diff
@@ -18,6 +18,26 @@ type Route struct {
...
+	return r.path
+}
+
+// GetMethod returns route method value (GET/POST/...).
+func (r Route) GetMethod() string {
```
Comment (mantzas) [inline component/http/route.go]:
```suggestion
func (r Route) Method() string {
```

File: component/http/route.go [hunk 3]
```diff
@@ -18,6 +18,26 @@ type Route struct {
...
+	return r.method
+}
+
+// GetMiddlewares returns route middlewares.
+func (r Route) GetMiddlewares() []MiddlewareFunc {
```
Comment (mantzas) [inline component/http/route.go]:
```suggestion
func (r Route) Middlewares() []MiddlewareFunc {
```

File: component/http/route.go [hunk 4]
```diff
@@ -18,6 +18,26 @@ type Route struct {
...
+	return r.middlewares
+}
+
+// GetHandler returns route handler function.
+func (r Route) GetHandler() http.HandlerFunc {
```
Comment (mantzas) [inline component/http/route.go]:
```suggestion
func (r Route) Handler() http.HandlerFunc {
```

**Review by mantzas — CHANGES_REQUESTED** [review]

**Review by mantzas — APPROVED** [review]

**Review by drakos74 — APPROVED** [review]

**Pull request merged**